bonfire/config.py

```diff
@@ -54,17 +54,30 @@
 # set to true when bonfire is running via automation using a bot acct (not an end user)
 BONFIRE_BOT = os.getenv("BONFIRE_BOT")
 
-AUTO_ADDED_FRONTEND_DEPENDENCIES = (
+DEFAULT_FRONTEND_DEPENDENCIES = (
     "chrome-service",
     "frontend-configs",
     "rbac",
     "rbac-frontend",
     "host-inventory",
     "host-inventory-frontend",
-    "unleash-proxy"
+    "unleash-proxy",
 )
 
 
+def _parse_frontend_dependencies():
+    env_var = os.getenv("BONFIRE_FRONTEND_DEPENDENCIES")
+    if isinstance(env_var, str):
+        if env_var.strip() == "":
+            return set()
+        return set(env_var.split(","))
+
+    return set(DEFAULT_FRONTEND_DEPENDENCIES)
```

Comment:
I'm wondering, maybe it's an enhancement but we're changing the type of `AUTO_ADDED_FRONTEND_DEPENDENCIES` from `tuple` to `set` , as long as we're fine with it then OK.

Also, I think this needs to strip out the spaces in case someone uses spaces:

```
BONFIRE_FRONTEND_DEPENDENCIES='bar ,   baz   , foo' 
```
Also, since `os.getenv` always returns either a `None` if not defined, or a `str` if defined, one could rewrite as



```suggestion

def _parse_frontend_dependencies():
    env_var = os.getenv("BONFIRE_FRONTEND_DEPENDENCIES")
    result = set()
  
    if env_var is None:
        result = set(DEFAULT_FRONTEND_DEPENDENCIES)
    else:
        for app in (env_var.split(",")):
            result.add(app.strip())

    return result

    return set(DEFAULT_FRONTEND_DEPENDENCIES)
```

Reply:
Had to tweak slightly but adopted your general idea here